pkg/log/log.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package log contains logging related functionality
+package log
+
+import (
+	"io"
+)
+
+// Logger is a interface that should be used for all logging inside kind/pkg.
+// Consumers of kind/pkg such as kind/cmd should define an object that satisfies
+// this interface.
+type Logger interface {
+	Output() io.Writer
+	SetOutput(out io.Writer)
+
+	Error(...interface{})
+	Errorf(string, ...interface{})
+
+	Warning(...interface{})
+	Warningf(string, ...interface{})
+
+	Info(...interface{})
+	Infof(string, ...interface{})
+
+	Debug(...interface{})
+	Debugf(string, ...interface{})
+}
+
+var logger Logger
+
+// GetLogger returns the locally defined logger that was previously set using SetLogger().
+func GetLogger() Logger {
```

Comment:
unless we're wrapping this, maybe we should just make the variable public? /shrug

Reply:
public var is also OK.
the only benefits of these getter/setting here is that they leave the option to do extra work eventually without breaking the user.

Reply:
SGTM